File: Beginner/compressList/compressList.java
//run as java -ea compressList
import java.util.Scanner;

public class compressList{
	public static void main(String[] args){
		int T,N,j,count;
		Scanner scan = new Scanner(System.in);

		T = scan.nextInt();
		assert(T>=1 && T<=100) : "T must be 1 <= T <= 100";

		while(T>0){
			N = scan.nextInt();
			assert(N>=1 && N<=100) : "N must be 1 <= N <= 100";

			int[] A = new int[N];

			for(int i=0;i<N;i++){
				A[i] = scan.nextInt();
				assert(A[i]>=1 && A[i]<=1000) : "A["+i+"] must be 1 <= A[i] <= 1000";

				if(i>0){
					assert(A[i]>A[i-1]) : "A must be sorted";
				}
			}

			j = 0;
			System.out.print(A[0]);

			while(j<N-1){
				count = 0;

				while(j<N-1){
					if(A[j+1]!=A[j]+1){
						break;
					}

					j++;
					count++;
				}

				if(count==0){
					j++;
				}

				if(count>=2){
					System.out.print("..."+A[j]);
				}else{
					System.out.print(","+A[j]);
				}
			}

			System.out.println();

			T--; 
		}
	}
}

File: Beginner/compressList/compressList.c
#include <stdio.h>
#include <assert.h>

int main(void){
	int T,N,i,count;

	scanf("%d",&T);
	assert(T>=1 && T<=100);

	while(T>0){
		scanf("%d",&N);
		assert(N>=1 && N<=100);

		int A[N];

		for(int i=0;i<N;i++){
			scanf("%d",&A[i]);
			assert(A[i]>=1 && A[i]<=1000);

			if(i>0){
				assert(A[i]>A[i-1]);
			}
		}

		i = 0;
		printf("%d",A[0]);

		while(i<N-1){
			count = 0;

			while(i<N-1){
				if(A[i+1]!=A[i]+1){
					break;
				}

				i++;
				count++;
			}

			if(count==0){
				i++;
			}
			if(count>=2){
				printf("...%d",A[i]);
			}else{
				printf(",%d",A[i]);
			}
		}

		printf("\n");

		T--;
	}

	return 0;
}